### ISSUE TITLE
Weak Diffie-Hellman key exchange parameters

### ISSUE THREAD

**Issue by bitmand** [dokku/dokku#2984]
Apps using the Lets Encrypt Dokku plugin suffer from weak Diffie-Hellman (DH) key exchange parameters. A SSL Labs test results in grade B just because the server is setup using common DH primes:

<img width="969" alt="ssllabs" src="https://user-images.githubusercontent.com/344214/32777779-cdcb83ee-c937-11e7-9455-a4e32911598b.png">

As far as I can tell, the Lets Encrypt plugins does not handle nginx ssl settings, but instead relies on the Certs plugin to manage those. Although I haven't tested with the certs plugin, I would expect it to have the same problem.

This is on Ubuntu 16.04.

An example on how to fix, could be to once build a global custom dh parameters files with:

    openssl dhparam -out /etc/nginx/dhparam.pem 2048

And once include it in the default dokku nginx file `/etc/nginx/conf.d/dokku.conf` by adding this line:

    ssl_dhparam /etc/nginx/dhparam.pem;

I'll happily test, provide more info or anything else if needed.

dokku report results:
```
-----> memory:
              total        used        free      shared  buff/cache   available
Mem:           7980         477        2709          50        4793        7235
Swap:           511           0         511
-----> docker version:
Client:
 Version:      17.10.0-ce
 API version:  1.33
 Go version:   go1.8.3
 Git commit:   f4ffd25
 Built:        Tue Oct 17 19:04:16 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.10.0-ce
 API version:  1.33 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   f4ffd25
 Built:        Tue Oct 17 19:02:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
-----> docker daemon info:
Containers: 19
 Running: 6
 Paused: 0
 Stopped: 13
Images: 32
Server Version: 17.10.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 0351df1c5a66838d0c392b4ac4cf9450de844e2d
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.50-x86_64-linode86
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.793GiB
Name: 
ID: WWQU:HIVM:YFSD:IIXG:3F5K:FVAW:6ENZ:PGOE:II6W:PLXN:WYFI:7ONG
Docker Root Dir: /var/lib/docker
Debug Mode (client): true
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
-----> sigil version: 0.4.0
-----> herokuish version:
herokuish: 0.3.32
buildpacks:
  heroku-buildpack-multi     v1.0.0
  heroku-buildpack-ruby      v168
  heroku-buildpack-nodejs    v99
  heroku-buildpack-clojure   v77
  heroku-buildpack-python    v99
  heroku-buildpack-java      v53
  heroku-buildpack-gradle    v23
  heroku-buildpack-scala     v78
  heroku-buildpack-play      v26
  heroku-buildpack-php       v122
  heroku-buildpack-go        v72
  heroku-buildpack-erlang    fa17af9
  buildpack-nginx            v8
-----> dokku version: 0.10.5
-----> dokku plugins:
plugn: 0.3.0
  00_dokku-standard    0.10.5 enabled    dokku core standard plugin
  20_events            0.10.5 enabled    dokku core events logging plugin
  apps                 0.10.5 enabled    dokku core apps plugin
  build-env            0.10.5 enabled    dokku core build-env plugin
  certs                0.10.5 enabled    dokku core certificate management plugin
  checks               0.10.5 enabled    dokku core checks plugin
  common               0.10.5 enabled    dokku core common plugin
  config               0.10.5 enabled    dokku core config plugin
  copy-files-to-image  1.0.0 enabled    dokku plugin to copy files from the host to the built docker image
  docker-options       0.10.5 enabled    dokku core docker-options plugin
  domains              0.10.5 enabled    dokku core domains plugin
  enter                0.10.5 enabled    dokku core enter plugin
  git                  0.10.5 enabled    dokku core git plugin
  letsencrypt          0.8.6 enabled    Automated installation of let's encrypt TLS certificates
  logs                 0.10.5 enabled    dokku core logs plugin
  mysql                1.0.0 enabled    dokku mysql service plugin
  named-containers     0.10.5 enabled    dokku core named containers plugin
  nginx-vhosts         0.10.5 enabled    dokku core nginx-vhosts plugin
  plugin               0.10.5 enabled    dokku core plugin plugin
  proxy                0.10.5 enabled    dokku core proxy plugin
  ps                   0.10.5 enabled    dokku core ps plugin
  redis                1.0.0 enabled    dokku redis service plugin
  repo                 0.10.5 enabled    dokku core repo plugin
  shell                0.10.5 enabled    dokku core shell plugin
  ssh-keys             0.10.5 enabled    dokku core ssh-keys plugin
  storage              0.10.5 enabled    dokku core storage plugin
  tags                 0.10.5 enabled    dokku core tags plugin
  tar                  0.10.5 enabled    dokku core tar plugin

```

**Comment by josegonzalez** [comment]
Mind testing this with the certs plugin?

**Comment by bitmand** [comment]
@josegonzalez I have tried adding a new dummy app, with a domain and ran `dokku certs:generate` to enable SSL. It's the same `$APP/nginx.conf` as my app using LE (ssl settings). And there are not changes to the `/etc/nginx/conf.d/dokku.conf`.

I should be able to *fix* this by simply adding my custom SSL configuration to `$APP/nginx.conf.d/<somefile>`, but I would prefer that Dokku provided me with the best SSL configuration as possible out of the box :) ( it already only support TLSv1.2 👍 )

**Comment by josegonzalez** [comment]
Closed by #2985.
